stl/inc/bitset

```diff
@@ -132,7 +132,7 @@ private:
     static constexpr unsigned long long _Mask = (1ULL << (_Need_mask ? _Bits : 0)) - 1ULL;
 
 public:
-    _NODISCARD constexpr bool operator[](size_t _Pos) const noexcept /* strengthened */ {
+    _NODISCARD constexpr bool operator[](const size_t _Pos) const noexcept /* strengthened */ {

```

Comment:
This is pass by value.  Is the `const` going to change behavior in any way?

Reply:
No, this change is completely unobservable to callers. WG21-N5001 \[dcl.fct\]/3:

> After producing the list of parameter types, any top-level *cv-qualifier*s modifying a parameter type are deleted when forming the function type.

Top-level `const` on value parameters is near-immediately ignored for function declarations. For function definitions, it affects the implementation, but in both cases it doesn't affect callers and doesn't affect the function type. (So the function's mangled name is unchanged, and this is ABI-invisible.)

We routinely add top-level `const` to STL functions and this has never caused trouble.

Reply:
I agree that this won't cause any trouble.  I was moreso asking if there's any purpose/benefit.  It prevents you from reusing the input parameter as another local variable but that's rarely something anyone tries to do.

Reply:
@dmachaj You can consider this as yet another function-member-variable. It just happened to be declared in the function argument list. But the rules of `const/non const` absolutely the same as for variables within the function.
This approach is used throughout the STL.

Reply:
> It prevents you from reusing the input parameter as another local variable but that's rarely something anyone tries to do.

It's low risk, but it's also cheap to prevent. There are two other reasons that we love marking value parameters and local variables as `const` whenever possible:

* It allows us to avoid worrying about whether the state will later change.
* When consistently followed (we aren't perfect about this but we try, especially in new code), the *lack* of `const` begins to become a signal that the parameter/variable's value will be altered later, so we should pay special attention to it.